### PR TITLE
Character aliasing

### PR DESCRIPTION
**Pull request by Marri**
Remaining:
- [x] Write tests
- [x] Add alias to API endpoint join
- [x] Include in Firefox refresh hack
- [x] Add a way to create, update and delete aliases
- [x] Rename stash and rebase it up
- [x] Default to most recent?
- [x] Display in profile somewhere
- [x] ~~Replace template_name with this intelligently?~~ Follow up later, this will be more work

Fixes issue #49

### CODE REVIEW

File: app/assets/javascripts/posts.js [hunk 1]
```diff
@@ -111,9 +116,11 @@ $(document).ready(function() {
   var displayCharID = $("#post-editor .post-character").data('character-id');
   var selectedIconID = $("#reply_icon_id").val();
   var displayIconID = $("#current-icon").data('icon-id');
+  var selectedAliasID = $("#reply_character_alias_id").val();
+  var displayAliasID = $("#post-editor .post-character").data('alias-id');
   if (selectedCharID != displayCharID) {
-    getAndSetCharacterData(selectedCharID, {restore_icon: true});
-    $("#active_character").val(selectedCharID).trigger("chosen:updated");
```
Comment (Marri) [inline app/assets/javascripts/posts.js]: We don't use chosen any more, so I took out the trigger

File: app/views/posts/_editor.haml [hunk 2]
```diff
@@ -14,13 +14,18 @@
       - elsif current_user.avatar
         = select_tag :icon_dropdown, options_for_select([[current_user.avatar.keyword, current_user.avatar.id]], current_user.avatar.id), prompt: "No Icon"
     .post-info-text
-      .post-character{class: ('hidden' unless item.character), data: {character_id: item.character_id}}= item.character.try(:name)
+      .post-character{class: ('hidden' unless item.character), data: {character_id: item.character_id || '', alias_id: item.character_alias_id || ''}}
```
Comment (Marri) [inline app/views/posts/_editor.haml]: These must default to '' or the attribute can be left out of the element entirely, which might break the javascript?

Comment (Throne3d) [inline app/views/posts/_editor.haml]: Pretty sure `nil` in the data gets made into a blank string – I at least didn't have trouble at any point that I tested it with the previous setup – but I'm not 100% sure.

**Comment by Marri** [comment]
It was not doing the thing at least locally for me until I used '' instead
of nil. Might be a browser thing? Meh, whatev, it's present now.

**Pull request merged**